internal/settings/providers/netcup/dnsrecord.go

```diff
@@ -0,0 +1,48 @@
+package netcup
+
+type DNSRecord struct {
+	ID           string `json:"id"`
+	DeleteRecord bool   `json:"deleterecord"`
+	Destination  string `json:"destination"`
+	Hostname     string `json:"hostname"`
+	Priority     string `json:"priority"`
+	State        string `json:"state"`
+	Type         string `json:"type"`
+}
+
+func NewDNSRecord(hostname, dnstype, destination string) *DNSRecord {
+	return &DNSRecord{
+		Destination: destination,
+		Hostname:    hostname,
+		Type:        dnstype,
+	}
+}
+
+type DNSRecordSet struct {
+	DNSRecords []DNSRecord `json:"dnsrecords"`
+}
+
+func NewDNSRecordSet(records []DNSRecord) *DNSRecordSet {
+	return &DNSRecordSet{
+		DNSRecords: records,
+	}
+}
+
+func (r *DNSRecordSet) GetRecord(name, dnstype string) *DNSRecord {
+	for _, record := range r.DNSRecords {
+		if record.Hostname == name && record.Type == dnstype {
+			return &record
```

Comment:
return a value, not a ptr, there is no need for a ptr

```suggestion
func (r *DNSRecordSet) GetRecord(name, dnstype string) DNSRecord {
	for _, record := range r.DNSRecords {
		if record.Hostname == name && record.Type == dnstype {
			return record
```